subsys/usb/Kconfig

```diff
@@ -114,6 +114,38 @@ config USB_MAX_POWER
 	  Set bMaxPower value in the Standard Configuration Descriptor,
 	  the result is 2mA times the value provided.
 
+config USB_WORKQUEUE
+	bool "Use a dedicate work queue in the USB subsystem"
+	default y if USB_CDC_ACM
```

Comment:
Shouldn't we limit this a bit? I think there is no good reason to force dedicated work queue for all CDC_ACM samples and increase RAM usage if could be avoided. The issues are visible when CDC_ACM is used as a backed for other subsystems which also use system work q for scheduling their work. I think this, at least for the time being, should be enabled only if `CONFIG_MCUMGR_ON_DEV_NAME="CDC_ACM_n"`, where n - number of CDC_ACM device. The problem is (afaik) we are unable to compare strings within Kconifgs. The idea of how to address such problem I described in [comment](https://github.com/zephyrproject-rtos/zephyr/pull/28339#issuecomment-691927108) for #28339. The question is if we are okay with accepting increasing the resources needed or not, if yes, then skip this comment.

Reply:
> Shouldn't we limit this a bit?

Well it is not "forced", you can still disable it. I think it should be `default y if USB_CDC_ACM`

Reply:
I don't want things that work with "real" (threadless) UARTs to break when switched to CDC-ACM.  IMO this is the safest way to do that.  We already know shell with bluetooth, and mcumgr, independently run into this problem.

We can take this out, but then when people expand a working sample to add code that starts to use the system work queue it will stop working, and much time will be wasted figuring out why.